### PR TITLE
Technical/player spawn

### PR DESCRIPTION
**Pull request by psoap**

### CODE REVIEW

File: excitebike/src/test/java/com/codenjoy/dojo/excitebike/model/PlayersSpawnSystemParametrizedTest.java [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+        //when
+        for (int i = 0; i < newPlayerNumberAfterInit; i++) {
+            Game game = createNewGame(field, factory);
+            games.add(game);
+        }
+        String res = (String) games.get(0).getBoardAsString();
+
+        //then
+        assertEquals(expected, res);
```
Comment (Logrus86) [inline excitebike/src/test/java/com/codenjoy/dojo/excitebike/model/PlayersSpawnSystemParametrizedTest.java]: Let's use hamcrest matcher assertThat instead, it shows really more explanatory failure message

File: excitebike/src/test/java/com/codenjoy/dojo/excitebike/model/PlayersSpawnSystemParametrizedTest.java [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+        Dice dice = mock(Dice.class);
+        when(dice.next(anyInt())).thenReturn(5);
+        GameField field = new GameFieldImpl(mapParser, dice);
+        PrinterFactory factory = new PrinterFactoryImpl();
+
+        List<Game> games = new ArrayList<>();
+
+        if (mapParser.getBikes().isEmpty()) {
+            int playersNumber = StringUtils.countMatches(expected, BikeType.OTHER_BIKE.ch()) + 1;
+            for (int i = 0; i < playersNumber; i++) {
```
Comment (Logrus86) [inline excitebike/src/test/java/com/codenjoy/dojo/excitebike/model/PlayersSpawnSystemParametrizedTest.java]: Why not 'forEach'? It's shorter usually :)

File: excitebike/src/test/java/com/codenjoy/dojo/excitebike/model/PlayersSpawnSystemParametrizedTest.java [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+                Game game = createNewGame(field, factory);
+                games.add(game);
+            }
+        } else {
+            for (Bike bike : mapParser.getBikes()) {
```
Comment (Logrus86) [inline excitebike/src/test/java/com/codenjoy/dojo/excitebike/model/PlayersSpawnSystemParametrizedTest.java]: Why not 'forEach'? It's shorter usually :)

File: excitebike/src/test/java/com/codenjoy/dojo/excitebike/model/PlayersSpawnSystemParametrizedTest.java [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+                Game game = createNewGame(field, factory);
+                games.add(game);
+                ((Player) game.getPlayer()).setHero(bike);
+            }
+        }
+
+        //when
+        for (int i = 0; i < newPlayerNumberAfterInit; i++) {
```
Comment (Logrus86) [inline excitebike/src/test/java/com/codenjoy/dojo/excitebike/model/PlayersSpawnSystemParametrizedTest.java]: Why not 'forEach'? It's shorter usually :)

File: excitebike/src/test/java/com/codenjoy/dojo/excitebike/model/PlayersSpawnSystemParametrizedTest.java [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+                                "       " +
+                                "       " +
+                                "       " +
+                                "       " +
+                                "       " +
+                                "■■■■■■■",
+                        "■■■■■■■\n" +
+                                "       \n" +
+                                "       \n" +
+                                " e     \n" +
```
Comment (Logrus86) [inline excitebike/src/test/java/com/codenjoy/dojo/excitebike/model/PlayersSpawnSystemParametrizedTest.java]: But it's not the first column, it starts from second column, isn't it?
Here and in every other case.
Could we fix it?

File: excitebike/src/main/java/com/codenjoy/dojo/excitebike/model/GameFieldImpl.java [hunk 6]
```diff
@@ -111,13 +115,78 @@ public boolean isDownLineChanger(int x, int y) {
...
+    }
+
+    private boolean tryToSetFreeCoordinates(Bike bike, Map<Integer, Long> bikesOnYCount, int x){
+        List<Integer> yCoordinatesSortedByBikesCount = bikesOnYCount.entrySet()
+                .stream()
+                .sorted(comparingByValue())
+                .map(Map.Entry::getKey)
+                .collect(Collectors.toList());
+
+        for (Integer possibleY : yCoordinatesSortedByBikesCount) {
```
[the remark this answers is not in the document]
Comment (psoap) [reply excitebike/src/main/java/com/codenjoy/dojo/excitebike/model/GameFieldImpl.java]: Found shorter solution by using anyMatch() method after map()

**Pull request merged**